mne_bids/mne_bids.py

```diff
@@ -659,7 +646,7 @@ def write_raw_bids(raw, bids_fname, output_path, events_data=None,
     else:
         sh.copyfile(raw_fname, bids_fname)
     # KIT data requires the marker file to be copied over too
-    if hpi is not None:
+    if raw.info['dig'] is not None:
```

Comment:
@monkeyman192 I started rewriting this but I think I need your opinion before I jump right in. Why is the `_marker` file for KIT data not stored in the same folder as the raw data -- like in the case of CTF? I think MNE-BIDS is not for the case where files are scattered all around. It's more when files are already semi-organized but it's not necessarily BIDS format.

What we could do is to reverse engineer the [read_mrk](https://github.com/mne-tools/mne-python/blob/e4c995b4517089f7f105a9afb997f1b05a4e3653/mne/io/kit/coreg.py#L16) function that @teonbrooks wrote in MNE and write out the `sqd` marker file from that. We do not need to worry about what format the file was originally in because as we were discussing in the BIDS specification github page, only one filetype per manufacturer is going to be supported for these kinds of files.

Let me know what you think.

Reply:
the marker file is stored in the same location...
The sub folder with the same name as the sidecar.json contains the .mrk and .con file:
![image](https://user-images.githubusercontent.com/8000597/47050613-c9078380-d1ec-11e8-9cf3-038d4cfa7eef.png)
or do you mean before exporting?
Generally the .mrk file *should* be in the same folder as the .con file, but it doesn't have to, nor will it necessarily have the same name.

Also, doing the reverse of that function you will lose a vast amount of data. As you can see in that picture above the .mrk files we have are ~3Mb, even though most of it isn't used.

Reply:
Hmm okay I see. If it’s KIT specific issue, we should probably keep it out of MNE-BIDS for now then. We should go by 80-20 philosophy.

Reply:
> since not being able to handle the markers is the same as not being able to handle the headshape/electrode placement files also

we won't have this issue of losing information with the headshape files, no? They are simple text files and we can write them out. This is only an issue with `hpi` as far as I understand.

Reply:
Ah yeah, that's true. Yeah, hpi only seems to be used for the KIT data also. Really I feel like it should be something that is stored in the Raw object and extracted if need-be (like even just store the path in the `raw.filenames` property).
If the value was stored here, then this would become a non-issue.

Reply:
I mean you could in principle store it as a private attribute of `raw`. I think nothing stops us from doing that in our `read_raw` function. But it feels a bit of a hack to me, so I'd wait a bit before going that way ...

Reply:
yeah, that's why I propose adding it to the `raw.filenames` property. It's already a list, and for KIT data at least the .mrk file is necessary. It doesn't seem *too* unreasonable for the marker file to be listed there too...
You could also maybe just add it to the `raw._raw_extras` dictionary.

Reply:
yes, I agree that makes sense ...

Reply:
`hpi` is an argument of `read_raw` function. the bidsification could happen there. I think it is a blocker if we don't accept hpi files because that breaks pipelines for source modeling for non FIF and CTF systems

Reply:
alright fixed. It works now as long as you use our general reader :)